src/apix-components/apix-api.service.ts

```diff
@@ -0,0 +1,287 @@
+            .then(response => response.json())
+            .catch(this.handleError);
+    }
+
+    getLocalApis() {
+        return this.http.get(`${this.localApiUrl}`)
+            .toPromise()
+            .then(response => response.json())
+            .catch(this.handleError);
+    }
+
+    getRemoteApi(id: number): Promise<any> {
+        return this.http.get(`${this.remoteApiUrl}/apis/${id}`)
+               .toPromise()
+               .then(response => response.json())
+               .catch(this.handleError);
+    }
+
+    getLatestRemoteApiIdForApiUid (api_uid: string) {
+        return this.http.get(`${this.remoteApiUrl}/apis/uids/${api_uid}`)
+            .toPromise()
+            .then(response => response.json())
+            .catch(this.handleError);
+
+    }
+
+    getRemoteApiResources (apiId: number) {
+        return this.http.get(`${this.remoteApiUrl}/apis/${apiId}/resources`)
+            .toPromise()
+            .then(response => response.json())
+            .catch(this.handleError);
+    }
+
+    getHTMLResponse (url: string) {
+        return this.http.get(`${url}`)
+            .toPromise()
+            .then(response => response)
+            .catch(this.handleError);
+    }
+
+    getJSONResponse (url: string) {
+        return this.http.get(`${url}`)
+            .toPromise()
+            .then(response => response.json())
+            .catch(this.handleError);
+    }
+
+    getSwaggerConsoleHTML (url: string) {
+        return this.http.get(`${url}`)
+            .toPromise()
+            .then(response => response)
+            .catch(this.handleError);
+    }
+
+    getSamples (platform: string) {
+        if (!platform) {
+            return;
+        }
+
+        // aaron note: it seems that the apigw can only support the syntax of having a single instance of
+        // a given query argument, so insteand of multiple platform values, pass all values comma separated.
+        var url = this.remoteSXApiUrl + '/search/samples?platform=' + encodeURIComponent(platform) + '&summary=true';
+        return this.http.get(url)
+            .toPromise()
+            .then(response => response.json())
+            .catch(this.handleError);
+    }
+
+    /**
+     * Get the local API group overview path
+     */
+    getLocalAPIGroupOverviewPath (path: string) {
+        return this.http.get(`${this.localApiUrl}/${path}`)
+            .toPromise()
+            .then(response => response.json())
+            .catch(this.handleError);
+    }
+
+    login(username: string, password: string, authUrl: string) : Promise<any> {
+        return this.http.post(authUrl + '/login', {user: username, password: password})
+               .toPromise()
+               .then(response => {
+                   let user = response.json();
+                   console.log(user);
+                   this.addUserToStorage(user);
+               });
+    }
+
+    vcenterLogin(username: string, password: string, auth: Auth) : Promise<any> {
```

Comment:
I really don't like having the SSO coupled into this file, please put some TODO comment in here about needing to refactor to move it out.

Reply:
I have created a new ApixAuthService.ts to decouple the auth logic from the APIs.